expression/aggregation/aggregation_test.go

```diff
@@ -46,57 +45,62 @@ func generateRowData() []chunk.Row {
 	return rows
 }
 
-func (s *testAggFuncSuit) SetUpSuite(c *C) {
+func createmockAggFuncSuite(t *testing.T) (s *mockAggFuncSuite) {
+	s = new(mockAggFuncSuite)
 	s.ctx = mock.NewContext()
 	s.ctx.GetSessionVars().GlobalVarsAccessor = variable.NewMockGlobalAccessor()
 	s.rows = generateRowData()
 	s.nullRow = chunk.MutRowFromDatums([]types.Datum{{}}).ToRow()
+	return
 }
 
-func (s *testAggFuncSuit) TestAvg(c *C) {
+func TestAvg(t *testing.T) {
+	s := createmockAggFuncSuite(t)
 	col := &expression.Column{
 		Index:   0,
 		RetType: types.NewFieldType(mysql.TypeLonglong),
 	}
 	ctx := mock.NewContext()
 	desc, err := NewAggFuncDesc(s.ctx, ast.AggFuncAvg, []expression.Expression{col}, false)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	avgFunc := desc.GetAggFunc(ctx)
 	evalCtx := avgFunc.CreateContext(s.ctx.GetSessionVars().StmtCtx)
 
 	result := avgFunc.GetResult(evalCtx)
-	c.Assert(result.IsNull(), IsTrue)
+	require.True(t, result.IsNull())
 
 	for _, row := range s.rows {
 		err := avgFunc.Update(evalCtx, s.ctx.GetSessionVars().StmtCtx, row)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}
 	result = avgFunc.GetResult(evalCtx)
 	needed := types.NewDecFromStringForTest("67.000000000000000000000000000000")
-	c.Assert(result.GetMysqlDecimal().Compare(needed) == 0, IsTrue)
+	require.True(t, result.GetMysqlDecimal().Compare(needed) == 0)
 	err = avgFunc.Update(evalCtx, s.ctx.GetSessionVars().StmtCtx, s.nullRow)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	result = avgFunc.GetResult(evalCtx)
-	c.Assert(result.GetMysqlDecimal().Compare(needed) == 0, IsTrue)
+	require.True(t, result.GetMysqlDecimal().Compare(needed) == 0)
 
 	desc, err = NewAggFuncDesc(s.ctx, ast.AggFuncAvg, []expression.Expression{col}, true)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	distinctAvgFunc := desc.GetAggFunc(ctx)
 	evalCtx = distinctAvgFunc.CreateContext(s.ctx.GetSessionVars().StmtCtx)
 	for _, row := range s.rows {
 		err := distinctAvgFunc.Update(evalCtx, s.ctx.GetSessionVars().StmtCtx, row)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}
 	result = distinctAvgFunc.GetResult(evalCtx)
 	needed = types.NewDecFromStringForTest("50.500000000000000000000000000000")
-	c.Assert(result.GetMysqlDecimal().Compare(needed) == 0, IsTrue)
+	require.True(t, result.GetMysqlDecimal().Compare(needed) == 0)
 	partialResult := distinctAvgFunc.GetPartialResult(evalCtx)
-	c.Assert(partialResult[0].GetInt64(), Equals, int64(100))
+	require.Equal(t, int64(100), partialResult[0].GetInt64())
 	needed = types.NewDecFromStringForTest("5050")
-	c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))
+	require.True(t, partialResult[1].GetMysqlDecimal().Compare(needed) == 0)
+	//c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))
```

Comment:
@tisonkun Where to log the following comment? 
`Commentf("%v, %v ", result.GetMysqlDecimal(), needed)`

Reply:
`c.Assert(partialResult[1].GetMysqlDecimal().Compare(needed) == 0, IsTrue, Commentf("%v, %v ", result.GetMysqlDecimal(), needed))`

can be almost

`require.Equalf(t, 0, partialResult[1].GetMysqlDecimal().Compare(needed), "%v, %v ", result.GetMysqlDecimal(), needed)`